src/GovUk.Education.ExploreEducationStatistics.Admin/Controllers/Api/ReleaseController2.cs

```diff
@@ -0,0 +1,117 @@
+    {
+        private readonly IReleaseService2 _releaseService2;
+        private readonly IPublicationService _publicationService;
+
+        public ReleasesController2(IReleaseService2 releaseService2,
+             IPublicationService publicationService)
+        {
+            _releaseService2 = releaseService2;
+            _publicationService = publicationService;
+        }
+
+        // POST api/publication/{publicationId}/releases
+        [HttpPost("publications/{publicationId}/releases")]
+        [AllowAnonymous] // TODO revisit when authentication and authorisation is in place
+        public async Task<ActionResult<ReleaseViewModel>> CreateReleaseAsync(CreateReleaseViewModel release,
+            PublicationId publicationId)
+        {
+            return await CheckPublicationExistsAsync(publicationId, () =>
+            {
+                release.PublicationId = publicationId;
+                return _releaseService2.CreateReleaseAsync(release);
+            });
+        }
+        
+        [HttpGet("releases/{releaseId}")]
+        [AllowAnonymous] // TODO revisit when authentication and authorisation is in place
+        public async Task<ReleaseViewModel> GetReleaseAsync(ReleaseId releaseId)
+        {
+            return await _releaseService2.GetReleaseForIdAsync(releaseId);
+        }
+
+        [HttpGet("releases/{releaseId}/summary")]
+        [AllowAnonymous] // TODO revisit when authentication and authorisation is in place
+        public async Task<ActionResult<EditReleaseSummaryViewModel>> GetReleaseSummaryAsync(ReleaseId releaseId)
+        {
+            return Ok(await _releaseService2.GetReleaseSummaryAsync(releaseId));
+        }
+
+        [HttpPut("releases/{releaseId}/summary")]
+        [AllowAnonymous] // TODO revisit when authentication and authorisation is in place
+        public async Task<ActionResult<ReleaseViewModel>> EditReleaseSummaryAsync(EditReleaseSummaryViewModel model,
+            ReleaseId releaseId)
+        {
+            return await CheckReleaseExistsAsync(releaseId, () =>
+            {
+                model.Id = releaseId;
+                return _releaseService2.EditReleaseSummaryAsync(model);
+            });
+        }
+        
+        [HttpGet("publications/{publicationId}/releases")]
+        [AllowAnonymous] // TODO We will need to do Authorisation checks when we know what the permissions model is.
+        public async Task<ActionResult<List<ReleaseViewModel>>> GetReleaseForPublicationAsync(
+            [Required] PublicationId publicationId)
+        {
+            return Ok(await _releaseService2.GetReleasesForPublicationAsync(publicationId));
+        }
+        
+        private async Task<ActionResult> CheckReleaseExistsAsync<T>(ReleaseId releaseId,
+            Func<Task<Either<ValidationResult, T>>> andThen)
+        {
+            var release = await _releaseService2.GetAsync(releaseId);
+            if (release == null)
```

Comment:
think we can use release.IsNull() here but as I said to be this is minor and probs just a std we need to agree - can go back and change things later.

Reply:
Done